File: Chapter 03/JS/section_3_3_3_lsp/after/coach.js
/**
 * THE CONFIDENT CONSUMER.
 * * ARCHITECTURE NOTE: Because we strictly adhered to LSP, the Coach class 
 * becomes incredibly simple. We no longer need 'if' checks to see if a 
 * player is a Goalie, nor do we need to wrap calls in try/catch blocks 
 * to handle unexpected behavior. 
 * * The Coach trusts the abstraction because the architecture is now honest.
 */
class Coach {
    /**
     * Directs the player to take their field position with absolute certainty.
     * @param {Player} fieldPlayer - A guaranteed field-capable athlete.
     */
    directFieldPlay(fieldPlayer) {
        console.log("  [Coach] Alright player, execute your field assignment!");
        
        // This call is now safe, predictable, and architecturally sound.
        fieldPlayer.playFieldPosition();
    }
}

module.exports = Coach;

File: Chapter 03/Python/section_3_3_3_lsp/after/coach.py
from .player import Player

"""
THE CONFIDENT CONSUMER.
* ARCHITECTURE NOTE: Because we strictly adhered to LSP, the Coach class 
becomes incredibly simple. We no longer need 'if' checks to see if a 
player is a Goalie (isinstance checks), nor do we need to worry about 
unexpected logic failures. 
* The Coach trusts the abstraction because the architecture is now honest.
"""
class Coach:
    def direct_field_play(self, field_player: Player):
        """
        Directs the player to take their field position with absolute certainty.
        """
        print("  [Coach] Alright player, execute your field assignment!")
        
        # This call is now safe, predictable, and architecturally sound.
        field_player.play_field_position()